test/test_helper.rb

```diff
@@ -20,27 +20,35 @@
 require 'fileutils'
 FileUtils.mkdir_p(File.expand_path('../../tmp/cache', __FILE__))
 
+# https://github.com/seattlerb/minitest/blob/master/lib/minitest/autorun.rb
 gem 'minitest'
-require 'minitest/autorun'
-require 'minitest/reporters'
-Minitest::Reporters.use!
-if defined?(Minitest::Test)
-  $minitest_version = 5 # rubocop:disable Style/GlobalVars
-  # Minitest 5
-  # https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest/autorun.rb
-  # https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest.rb#L45-L59
-else
-  $minitest_version = 4 # rubocop:disable Style/GlobalVars
+begin
+  require 'minitest'
+rescue LoadError
+  # Minitest 4
+  require 'minitest/unit'
+  require 'minitest/spec'
+  require 'minitest/mock'
+  $minitest_version = 4
   # Minitest 4
   # https://github.com/seattlerb/minitest/blob/644a52fd0/lib/minitest/autorun.rb
   # https://github.com/seattlerb/minitest/blob/644a52fd0/lib/minitest/unit.rb#L768-L787
   # Ensure backward compatibility with Minitest 4
   Minitest = MiniTest unless defined?(Minitest)
   Minitest::Test = MiniTest::Unit::TestCase
-  def Minitest.after_run(&block)
-    MiniTest::Unit.after_tests(&block)
-  end
+  minitest_run = ->(argv) { MiniTest::Unit.new.run(argv) }
+else
+  # Minitest 5
+  $minitest_version = 5
+  # Minitest 5
+  # https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest/autorun.rb
+  # https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest.rb#L45-L59
+  require 'minitest/spec'
+  require 'minitest/mock'
+  minitest_run = ->(argv) { Minitest.run(argv) }
 end
+require 'minitest/reporters'
+Minitest::Reporters.use!
```

Comment:
I think @zenspider said he doesn't like Minitest::Reporters, but I don't recall why.  It makes controlling the output(s) a lot easier :).  

(@zenspider this basically let's me guarantee I can run code after minitest finishes by moving the autorun from `at_exit` to `END`, which resolves before `at_exit` resolves.  This is sort of what I was talking about at RubyConf, that I couldn't run code around minitest.  This is Just Ruby, right? :)